Src/FluentAssertions/Equivalency/Steps/EnumerableEquivalencyStep.cs

```diff
@@ -70,8 +71,8 @@ internal static object[] ToArray(object value)
 
     private static bool IsIgnorableArrayLikeType(object value)
     {
-        return value.GetType() is Type type &&
-                    (type.Name.Equals("ImmutableArray`1", StringComparison.Ordinal) ||
-                     (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(ArraySegment<>)));
+        var type = value.GetType();
+        return  (type.Name.Equals("ImmutableArray`1", StringComparison.Ordinal) ||
+                (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(ArraySegment<>)));
```

Comment:
I think this does not satisfy the qodana report, does it?

Reply:
It asked me to use the is not null pattern. However, that does not work with implicit variable declration afaik. 

It was complaining about the:

someNullable is SomeType value

Pattern (as far as I understood at least)

Reply:
I think qodana means something like this:

```csharp
return value.GetType() is { } type && (type.Name.Equals("ImmutableArray`1", StringComparison.Ordinal) ||
        (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(ArraySegment<>)));
```

Reply:
`is not null`, right?

Reply:
Pushed a fix.

Reply:
And thanks.